value_matcher.go

```diff
@@ -116,8 +117,8 @@ func (m *valueMatcher) addTransition(val typedVal) *fieldMatcher {
 			var newNfa *smallTable[*nfaStepList]
 			newNfa, nextField = makeShellStyleAutomaton(valBytes, nil)
 			newDfa = nfa2Dfa(newNfa)
-		default:
```

Comment:
Why did you remove all the `default`s?

Reply:
You could test it IMHO (if you're concerned about coverage): create a new `valType` in your test and pass it. It should trigger the panic (which you can recover from and assert). Reason I like `panic` in those `switch` statements is that with the code base growing, slipping in a new `valType` and forgetting the places where this is `switch`-ed on is a possibility. Without a `default` case your code would happily continue and likely produce a bug.

Reply:
Update: here's an example how to start testing these cases

```go
func TestAddInvalidTransition(t *testing.T) {
	defer func() {
		if recover() == nil {
			t.Errorf("TestAddInvalidTransition should have panicked")
		}
	}()

	panicType := valType(999)

	m := newValueMatcher()
	v1 := typedVal{
		vType: panicType,
		val:   "one",
	}
	m.addTransition(v1)
}
```

Note: you could add more panic scenarios to cover the `if fields.startDfa != nil` branch. `default` in `singletonAutomaton` might not be needed as caught by earlier panic.

Reply:
Excellent, thanks, will do.

Reply:
This definitely improved both the code and the coverage, thanks.

Reply:
Awesome, glad to hear!